base_layer/core/src/base_node/states/listening.rs

```diff
@@ -118,11 +118,12 @@ impl ListeningInfo {
 
 // Finds the set of sync peers that have the best tip on their main chain.
 fn find_sync_peers(best_metadata: &ChainMetadata, metadata_rounds: &VecDeque<Vec<PeerChainMetadata>>) -> Vec<NodeId> {
-    let mut sync_peers = Vec::<NodeId>::new();
-    for peer_metadata in metadata_rounds.iter().flatten() {
-        if peer_metadata.chain_metadata == *best_metadata {
-            sync_peers.push(peer_metadata.node_id.clone());
-        }
-    }
-    sync_peers
+    let mut metadata_list = metadata_rounds.iter().flatten().collect::<Vec<_>>();
+    metadata_list.sort_by(|a, b| {
```

Comment:
These changes are not correct, two different competing chains might have the exact same accumulated difficulty. The find_sync_peers function should return only peers that have exactly the same chain.

Reply:
I thought so. Will make a change